libs/wingsdk/src/util/util.ts

```diff
@@ -57,6 +205,74 @@ export interface NanoidOptions {
  * Utility functions.
  */
 export class Util {
+  /**
+   * Execute a command in the shell and return its standard output.
+   * @param command - The shell command to execute.
+   * @param opts - The shell options including working directory, environment variables, etc.
+   * @returns The standard output of the shell command.
+   * @throws An error if the command returns a non-zero exit code.
+   */
+  public static async shell(
+    command: string,
+    opts?: ShellOptions
+  ): Promise<string> {
+    const args = command.split(" ");
+    const output = await Util.exec("/bin/sh", ["-c", ...args], opts);
+    if (output.status !== 0) {
+      throw new Error(
+        `Command exited with status ${output.status}: ${command}`
+      );
+    }
+    return output.stdout;
+  }
+
+  /**
+   * Execute a program with the given arguments, wait for it to finish, and
+   * return its outputs.
+   * @param program - The program to execute.
+   * @param args - An array of arguments to pass to the program.
+   * @param opts - Execution options including working directory, environment variables, etc.
+   * @returns A promise that resolves to the output of the executed program.
+   * @throws An error if the execution fails.
+   */
+  public static async exec(
+    program: string,
+    args: Array<string>,
+    opts?: ExecOptions
+  ): Promise<Output> {
+    const command = `${program} ${args.join(" ")}`;
+
+    try {
+      return {
+        stdout: execSync(command, opts).toString(),
+        stderr: "",
+        status: 0,
+      };
+    } catch (error: any) {
+      return {
+        stdout: error.stdout?.toString() ?? "",
+        stderr: error.stderr?.toString() ?? "",
+        status: error.status ?? -1,
+      };
+    }
```

Comment:
Let's see if we can use `exec` asynchronously... something like this perhaps

```js
import { exec as nodeExec } from "child_process";
import { promisify } from "util";

const exec = util.promisify(nodeExec);
const { stdout, stderr } = await exec(command, opts);
```

Reply:
Originally, I attempted to use a promisified version of Node.js's `child_process.exec()`, but I was unable to get the error code, so I eventually decided to switch to using `execSync()` instead.